contrib/pinot-druid-migration/src/main/java/org/apache/pinot/druid/tools/DruidToPinotSegmentConverter.java

```diff
@@ -0,0 +1,78 @@
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.druid.tools;
+
+import java.io.File;
+import org.apache.pinot.common.data.Schema;
+import org.apache.pinot.core.indexsegment.generator.SegmentGeneratorConfig;
+import org.apache.pinot.core.segment.creator.impl.SegmentIndexCreationDriverImpl;
+import org.apache.pinot.druid.data.readers.DruidSegmentRecordReader;
+
+
+/**
+ * The DruidToPinotSegmentConverter is a CLI tool that converts a Druid segment to a Pinot segment.
+ */
+public class DruidToPinotSegmentConverter {
+  private static String _schemaFilePath;
+  private static String _druidSegmentPath;
+  private static String _outPath;
+  private static String _segmentName;
+  private static String _tableName;
+
+  // TODO: Change implementation to use the Command framework like the CreateSegmentCommand
+  public static void convertSegment()
+      throws Exception {
+    File segment = new File(_druidSegmentPath);
+    Schema schema = Schema.fromFile(new File(_schemaFilePath));
+
+    final SegmentGeneratorConfig segmentGeneratorConfig = new SegmentGeneratorConfig();
+    segmentGeneratorConfig.setDataDir(_druidSegmentPath);
+    segmentGeneratorConfig.setOutDir(_outPath);
+    segmentGeneratorConfig.setOverwrite(true);
+    segmentGeneratorConfig.setTableName(_tableName);
+    segmentGeneratorConfig.setSegmentName(_segmentName);
+    segmentGeneratorConfig.setSchemaFile(_schemaFilePath);
+    segmentGeneratorConfig.setSchema(schema);
+
+    final SegmentIndexCreationDriverImpl driver = new SegmentIndexCreationDriverImpl();
+    SegmentGeneratorConfig config = new SegmentGeneratorConfig(segmentGeneratorConfig);
+    DruidSegmentRecordReader recordReader = new DruidSegmentRecordReader(segment, segmentGeneratorConfig.getSchema());
+    driver.init(config, recordReader);
+    driver.build();
+  }
+
+  public static void main(String[] args) {
+    if (args.length != 5) {
+      System.out.println("Convert a Druid segment into Pinot format.");
+      System.out.println("Use with arguments: <table-name> <segment-name> <segment-path> <schema-path> <output-directory>");
+    } else {
+      _tableName = args[0];
+      _segmentName = args[1];
+      _druidSegmentPath = args[2];
+      _schemaFilePath = args[3];
+      _outPath = args[4];
+
+      try {
+        convertSegment();
+      } catch (Exception e) {
+        throw new RuntimeException(e);
```

Comment:
Why do we catch exception and re-throw it as a runtime exception here?

Reply:
I saw it in a class that i was referencing so I guess i'll just take it out

Reply:
I think that this is not a good pattern. Put the following to the main function

```
public static void main(String[] args) throws Exception {
}
```